cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompiler.java

```diff
@@ -109,6 +116,19 @@ CompilationModellingResult compileSource(
     ExecutableModellingResult preCompileSource(SlangSource source, PrecompileStrategy precompileStrategy);
 
     /**
+     * Pre-compile a CloudSlang source into an {@link ExecutableModellingResult}.
+     * All errors that are found during pre-compilation are collected (an exception is not thrown)
+     *
+     * @param source             the {@link SlangSource}
+     * @param precompileStrategy whether to use caching in pre-compile.
+     * @param sensitivityLevel
+     * @return an {@link ExecutableModellingResult} object, containing an executable which is either a flow
+     *     or an operations in the file, and a list of all the errors that were found (if any).
+     */
+    ExecutableModellingResult preCompileSource(SlangSource source, PrecompileStrategy precompileStrategy,
```

Comment:
note: if I am using multi-line params, I like each param oin a new line like you did previously:
```java
    CompilationArtifact compile(
            SlangSource source,
            Set<SlangSource> path,
            PrecompileStrategy precompileStrategy,
            SensitivityLevel sensitivityLevel);
```